### DISCUSSION TITLE
New Language from Scratch

### DISCUSSION THREAD

**Discussion by skyecas** [Skiylia-Lang/VerboseLang#13] [General]
So: VerboseLang:
doesn't have an actual name, or anything really.

A lot of things need to be decided before any code can be laid down, such as:

- [ ] Naming:
  - [x] What is this language actually called?
  - [x] What file extension does it use?
- [ ] Syntax:
  - [ ] How does one declare variables
  - [ ] What about loops
  - [ ] or conditionals?
- [ ] Examples
  - [ ] Maybe a few examples of VerboseLang next to Python to highlight the difference
  - [ ] This also allows unit Testing to be performed
- [ ] Technical details
  - [x] Is this language [Compiled](https://en.wikipedia.org/wiki/Compiled_language), or [Interpreted](https://en.wikipedia.org/wiki/Interpreter_(computing)), or somewhere inbetween?
  - [ ] Dynamic or static typing?
  - [x] Bytecode, Tree-walk, or JIT?

And a whole host of other details I haven't thought to type yet.

Some resources to help with building this:
[Crafting Interpreters](http://craftinginterpreters.com/contents.html)
[My own (ongoing) creation](https://skiylia-lang.github.io/)
Any others welcome:

**Comment by skyecas** [comment]
More syntax ideas referenced in #3 #4 #5 #6 #7

**Comment by skyecas** [comment]
Could call it `Verboscript` with the `.vrbo` extension?

**Comment by Amritanshu02** [comment]
Hello! I would like to work upon this issue, could you suggest me where to start ?

**Comment by skyecas** [comment]
Well I'm currently looking through a lot of programing syntax and coming up with their equivalent in plain english, So if you wanted to suggest alternatives on any of their issues, or find new syntax examples not listed, that would probably be the best place to start.

There is a wiki page [here](https://github.com/Skiylia-Lang/Verboscript/wiki/Example-Syntax) of all the examples, though none have been decided concretely.

The project is still in it's infancy though, so all of the time spend is just planing and idea brainstorming.

**Comment by Nandan-D14** [comment]
hey ,  i would i like to work with this issue but i am new here can you please guide me
